internal/receiver/discoveryreceiver/README.md

```diff
@@ -0,0 +1,334 @@
+# Discovery Receiver
+
+| Status                   |                  |
+| ------------------------ | ---------------- |
+| Stability                | [in-development] |
+| Supported pipeline types | logs             |
+| Distributions            | [Splunk]         |
+
+The Discovery receiver is a logs pipeline compatible receiver that provides the ability to test the functional
+status of any receiver type whose target is reported by an Observer. It does this by providing configurable `status`
+match rules capable of evaluating the generated receiver's emitted metrics, if any, or component-level log statements
+(via the instance's [zap.Logger](https://pkg.go.dev/go.uber.org/zap)). It works similarly to the
+[Receiver Creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
+(in fact it wraps an internal instance of one) but the resulting dynamically-instantiated receivers don't actually
+report their metric content to your metrics pipelines. Instead the metrics are intercepted by an internal metrics
+consumer capable of translating desired metrics to log records based on `status: metrics` rules you define. All
+component-level log statements are similarly intercepted by a log evaluator and can be translated to emitted log records
+based on `status: statements` rules you define.
+
```

Comment:
Suggestion:

The Discovery receiver is a receiver compatible with logs pipeline which allows you to test the functional status of any receiver type whose target is reported by an Observer. It provides configurable `status`
match rules which evaluate the generated receiver's emitted metrics (if any), or component-level log statements, via the instance's [zap.Logger](https://pkg.go.dev/go.uber.org/zap). 

It works similarly to the [Receiver Creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/receivercreator/README.md)
(in fact, it wraps an internal instance of one), but the resulting dynamically-instantiated receivers don't actually
report their metric content to your metrics pipelines. Instead, the metrics are intercepted by an internal metrics
consumer capable of translating desired metrics to log records based on the `status: metrics` rules you define. All
component-level log statements are similarly intercepted by a log evaluator, and can be translated to emitted log records based on the `status: statements` rules you define.